### ISSUE TITLE
Opinion on Alternative Public Forks

### ISSUE THREAD

**Issue by addyess** [gtsystem/lightkube#49]
@gtsystem I wanted to start this issue by thanking you for the maintenance of an amazing package that helps those of us in @charmed-kubernetes to create easier-to-maintain projects

As more projects in @charmed-kubernetes make use of lightkube, we hope to insure our dependence on it by publicly forking the project and keeping it in sync with your [master](https://github.com/gtsystem/lightkube/tree/master) branch.  In the unfortunate event we must make a patch to lightkube quickly, and rebuild one of our projects -- without your upstream merge, we need a repository from which to build hot-patches.  

I wanted to get your blessing for this use case, and let you know we have no plans for creating an alternative project based on your work.  I'll encourage my team to use this new fork to raise PRs into this upstream source.  We may also create release branches in this fork in the event we need to backport a hot-patch into an older release of lightkube.

Thank you so much, please let me know if this will be an issue.

**Comment by gtsystem** [comment]
Hi @addyess, as far as I know there is already a fork from few years https://github.com/canonical/lightkube. Are you referring to this or you plan to create a new one?

Anyway I understand your requirement for business continuity and I don't have any concern. I will actually be happy if a couple of folks from your organization (maybe that have already contributed) can join as maintainers of this repo.  Would that be a better solution?
Also, how do you plan to consume the module in the case you need to make a release without me? Will you push a build in your private pipy repository?

**Comment by addyess** [comment]
@gtsystem 
yes, the plan is to keep https://github.com/canonical/lightkube in sync with your upstream master. Not a new one -- i'm trying to get better ownership over this fork ATM

I'm not sure who you'd choose to elect, but I would highly recommend @ca-scribner

I think if we had to create a hot-patch fix
* we'd create a branch in our fork
* apply the patch in the branch
* install directly from github via branch or commit hash
```
lightkube @ git+https://github.com/canonical/lightkube@41b95ec
... or ....
lightkube @ git+https://github.com/canonical/lightkube@patch-1.xx
```

**Comment by ca-scribner** [comment]
Yeah if interested I'm happy to help.  @gtsystem you have the vision for the project so things like feature requests should be your call, but if I can help (and provide a fail-safe for a quick critical bug fix) I'm happy to contribute

**Comment by gtsystem** [comment]
@ca-scribner, I invited you as collaborator for this repo, I'm not an expert in github workflow, but I believe this should give you rights to approve and merge.

**Comment by ca-scribner** [comment]
yes, looks like you're right!  Thanks.

I'll add email notifications for this repo so I'm a bit more up to date too

**Comment by gtsystem** [comment]
closing for now. If there is something else you would like to discuss, feel free to open a new issue or reach me directly.
